lib/decode/decode.go

```diff
@@ -120,15 +122,39 @@ func HookWeakDecodeFromSlice(from, to reflect.Type, data interface{}) (interface
 
 	switch d := data.(type) {
 	case []map[string]interface{}:
-		switch {
-		case len(d) == 0:
-			return nil, nil
-		case len(d) == 1:
-			return d[0], nil
-		default:
-			return data, nil
+		if len(d) == 1 {
+			return unSlice(d[0])
 		}
-	default:
-		return data, nil
+	// the JSON decoder can apparently decode slices as []interface{}
```

Comment:
I think it would be useful to have a test for this case, and also probably update the function comment to not explicitly reference `[]map[string]interface{}` anymore.

Reply:
Added a test case, and updated the docstring to mention both, and explain the `[]interface{}` case